eo-maven-plugin/src/main/java/org/eolang/maven/ChText.java

```diff
@@ -0,0 +1,116 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import java.nio.file.Path;
+import org.cactoos.Scalar;
+import org.cactoos.Text;
+import org.cactoos.io.InputOf;
+import org.cactoos.iterable.Filtered;
+import org.cactoos.scalar.FirstOf;
+import org.cactoos.scalar.Mapped;
+import org.cactoos.scalar.Unchecked;
+import org.cactoos.text.Flattened;
+import org.cactoos.text.Split;
+import org.cactoos.text.TextOf;
+
+/**
+ * Commit Hash from text.
+ *
+ * @since 0.28.11
+ */
+final class ChText implements CommitHash {
+
+    /**
+     * Commit Hash text source.
+     */
+    private final Scalar<String> source;
+
+    /**
+     * Tag.
+     */
+    private final String tag;
+
+    /**
+     * Production constructor.
+     *
+     * @param file Path to offline file with hashes and tags.
+     * @param tag Lookup tag.
+     */
+    ChText(final Path file, final String tag) {
+        this(() -> new TextOf(new InputOf(file)).asString(), tag);
+    }
+
+    /**
+     * The main constructor.
+     *
+     * @param source Text source.
+     * @param tag Lookup tag.
+     */
+    ChText(final Scalar<String> source, final String tag) {
+        this.source = source;
+        this.tag = tag;
+    }
+
+    @Override
+    public String value() {
+        return new Unchecked<>(
+            new Mapped<>(
+                Text::asString,
+                new FirstOf<>(
+                    new Split(
+                        new Flattened(
+                            new FirstOf<>(
+                                new Filtered<>(
+                                    t -> t.asString().contains(this.tag),
+                                    new Split(new TextOf(this.source), "\n")
+                                ),
+                                () -> {
+                                    throw new NotFound();
+                                }
+                            )
+                        ),
+                        "\\s+"
+                    ),
+                    () -> {
+                        throw new NotFound();
+                    }
+                )
+            )).value();
```

Comment:
@volodya-lombrozo check this: https://www.yegor256.com/2014/10/23/paired-brackets-notation.html

Reply:
@yegor256 solved.